doc/man7/migration_guide.pod

```diff
@@ -467,6 +467,17 @@ Remove your usage of the low level APIs. In this case you will need to rewrite y
 
 =back
 
+=head3 Error code changes
+
+As OpenSSL 3.0 provides a brand new Encoder/Decoder mechanism for working with
+widely used file formats, the application authors should consider changes the checks
+making on parsing. The password-protected keys in PKCS#8/PKCS#12 formats deserve a
+special attention. If only some errors are treated as indicator that the user
+slould be asked about the password again, it's worth adding processing the error
+lib B<ERR_LIB_OSSL_DECODER> and error reason B<ERR_R_UNSUPPORTED> as a new case.
```

Comment:
The ERR_R_UNSUPPORTED should not be returned for bad password case. So in the end I'd remove the whole sentence.

Reply:
@t8m, unfortunately it's the case I see in stack when no password is provided at all and the internal callback chain is used. See #16158

Reply:
Do you see this error with beta2?

Reply:
The last version I tested was beta1, I'll try to apply the changes in `crypto/encode_decode/decoder_lib.c` and check

Reply:
Just with the changes from `crypto/encode_decode/decoder_lib.c` I get the same result. I'll try to rebuild with the complete set of beta2 changes

Reply:
Full beta2 looks like returning UI_R_PROCESSING_ERROR

Reply:
Which makes some sense if the password used is empty.

Reply:
I.e., this is supposed to be returned if you use empty password or do not set any in callback but the key file needs a password to decrypt.